tests/unit/firebaseRef.spec.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+describe('firebaseRef', function () {
+
+  var firebaseRefProvider;
+  var MOCK_URL = 'https://stub.firebaseio-demo.com/'
+
+  beforeEach(module('firebase', function(_firebaseRefProvider_) {
+    firebaseRefProvider = _firebaseRefProvider_;
+  }));
+
+  describe('registerUrl', function() {
+
+    it('creates a single reference with a url', inject(function() {
+      firebaseRefProvider.registerUrl(MOCK_URL);
+      expect(firebaseRefProvider.$get()).toBeAFirebaseRef();
+    }));
+
+    it('creates a default reference with a config object', inject(function() {
+      firebaseRefProvider.registerUrl({
+        default: MOCK_URL
+      });
+      var firebaseRef = firebaseRefProvider.$get();
+      expect(firebaseRef).toBeAFirebaseRef();
+    }));
+
+    it('creates multiple references with a config object', inject(function() {
+      firebaseRefProvider.registerUrl({
+        default: MOCK_URL,
+        messages: MOCK_URL + 'messages'
```

Comment:
Ah the endless debate about whether we should suffix the root URL or concat all child paths with a separator. Reminds me of my old PHP days and little beauties like this:  `join(DIRECTORY_SEPARATOR, func_get_args($segments));`

There's nothing to see here. Move along.